IM_calculation/Advanced_IM/Models/Steel_MF_5Story/run.py

```diff
@@ -1,106 +1,144 @@
-    :return:
-    """
-    success_glob = os.path.join(output_dir, "Analysis_*")
-    success_files = glob.glob(success_glob)
-    model_converged = False
-    for f in success_files:
-        with open(f) as fp:
-            contents = fp.read()
-        model_converged = model_converged or (contents.strip() == "Successful")
-
-    im_csv_fname = os.path.join(output_dir, im_name + ".csv")
-    result_df = pd.DataFrame()
-    im_recorder_glob = os.path.join(output_dir, "env*/*.out")
-    im_recorders = glob.glob(im_recorder_glob)
-    value_dict = {}
-
-    for im_recorder in im_recorders:
-        im_name = os.path.splitext(os.path.basename(im_recorder))[0]
-        im_value = read_out_file(im_recorder, model_converged)
-
-        value_dict[im_name] = im_value
-
-    result_df = result_df.append(value_dict, ignore_index=True)
-    # print(result_df)
-    result_df.to_csv(im_csv_fname, index=False)
-
-
-def read_out_file(file, success=True):
-    if success:
-        with open(file) as f:
-
-            lines = f.readlines()
-
-            value = lines[-1].split()[1]
-
-            return value
-    else:
-        return float("NaN")
-
-
-if __name__ == "__main__":
-
-    main()
+"""
+Python script to run a 3D waveform through Steel_MF_5Story and store the outputs to a txt file
+"""
+
+import argparse
+import glob
+import os
+import subprocess
+
+import numpy as np
+import pandas as pd
+
+
+DEFAULT_OPEN_SEES_PATH = "OpenSees"
+
+model_dir = os.path.dirname(__file__)
+
+
+def main():
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "comp_000", help="filepath to a station's 000 waveform ascii file"
+    )
+    parser.add_argument(
+        "comp_090", help="filepath to a station's 090 waveform ascii file"
+    )
+    parser.add_argument(
+        "comp_ver", help="filepath to a station's ver waveform ascii file"
+    )
+    parser.add_argument(
+        "output_dir",
+        help="Where the IM_csv file is written to. Also contains the temporary recorders output",
+    )
+
+    parser.add_argument(
+        "--OpenSees_path",
+        default=DEFAULT_OPEN_SEES_PATH,
+        help="Path to OpenSees binary",
+    )
+
+    args = parser.parse_args()
+
+    output_dir = args.output_dir
+    if not os.path.exists(output_dir):
+        os.makedirs(output_dir)
+
+    script = [
+        args.OpenSees_path,
+        os.path.join(model_dir, "Run_script.tcl"),
+        args.comp_000,
+        output_dir,
+    ]
+
+    print(" ".join(script))
+
+    subprocess.call(script)
+
+    im_name = "Steel_MF_5Story"
+    create_im_csv(output_dir, im_name, "000")
+
+    script = [
+        args.OpenSees_path,
+        os.path.join(model_dir, "Run_script.tcl"),
+        args.comp_090,
+        output_dir,
+    ]
+
+    print(" ".join(script))
+
+    subprocess.call(script)
+
+    create_im_csv(output_dir, im_name, "090", print_header=False)
+
+    im_csv_fname = os.path.join(output_dir, im_name + ".csv")
```

Comment:
Why not passing im_csv_fname to create_im_csv? You are making up im_csv_fname there too.

Reply:
a few of them path&names can be centralized into qcore.sim_struct. so it can be reused and consistent across all codes. But since this is a merge into the advanced_im branch instead of the master; i wouldnt worry to much about it now; since i'm going to further modify the code from there.